### PR TITLE
Exception handling / it-academy.by / M-JC1-69-21 / Bozhen Davydov / Done tasks 1-7

### PR DESCRIPTION
**Pull request by davydovbv**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_01/Task01.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.rakovets.course.java.core.practice.exception_handling.task_01;
+
+public class Task01 {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_01/Task01.java:4]: Remove blank lines

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/task_01_Test/Task01Demo.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+public class Task01Demo {
+    public static void main(String[] args) {
+        try{
+            Task01.throwSomeException();
+        } catch (NullPointerException e) {
+            System.out.println("Field str of Object tsk1 is null!!!" );
+        }
+
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/task_01_Test/Task01Demo.java:13]: Remove blank line

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_03/Discount.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+public class Discount {
+
+    public static int provideDiscount(int sum) throws WrongSumException {
+        int result = sum;
+        if(sum > 100) {
+            result = (int)(sum*0.8);
+        } else if(sum <= 0) {
+            throw new WrongSumException();
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_03/Discount.java:11]: Contains violations of the Code of Convention

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/task_05_Test/Task05Demo.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+public class Task05Demo {
+    public static void main(String[] args) {
+       try {
+           Array.getFromArray(3);
+       } catch (Exception e){
+               throw new MyException(e.getMessage());
+           }
+       }
+    }
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/task_05_Test/Task05Demo.java:15]: Contains violations of the Code of Convention

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_06/Task06.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+    public  static int arrayOperations(int[] ar, int position, int number, int divider) throws InvalidNumber {
+        ar[position] = number / divider;
+        if(number % 2 !=0) {
+            throw new InvalidNumber();
+        }
+
+        return ar[position];
+    }
+}
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_06/Task06.java:13]: This solution is wrong

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_07/NumberChecker.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+public class NumberChecker {
+    public static int isEvenNumber(int number) throws NotEvenNumberException {
+        if(number % 2 == 0) {
+            System.out.print("Number is Even");
+            } else {
+                throw new NotEvenNumberException();
+            }
+
+        return number;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_07/NumberChecker.java:11]: Contains violations of the Code of Convention

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix the following problems

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_03/Discount.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.rakovets.course.java.core.practice.exception_handling.task_03;
+
+public class Discount {
+
+    public static int provideDiscount(int sum) throws WrongSumException {
+        int result = sum;
+        if(sum > 100) {
+            result = (int)(sum * 0.8);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_03/Discount.java:8]:
```suggestion
        if (sum > 100) {
            result = (int) (sum * 0.8);
```

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_07/NumberChecker.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.rakovets.course.java.core.practice.exception_handling.task_07;
+
+public class NumberChecker {
+    public static int isEvenNumber(int number) throws NotEvenNumberException {
+        if(number % 2 == 0) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_07/NumberChecker.java:5]:
```suggestion
        if (number % 2 == 0) {
```

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/task_01_Test/Task01Demo.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package com.rakovets.course.java.core.practice.exception_handling.task_01_Test;
+
+import com.rakovets.course.java.core.practice.exception_handling.task_01.Task01;
+
+public class Task01Demo {
+    public static void main(String[] args) {
+        try{
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/task_01_Test/Task01Demo.java:7]:
```suggestion
        try {
```

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_06/Task06.java (hunk 5 above)
Comment (rakovets) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/task_06/Task06.java:13]: Where is solution correct?

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix the following problems and add a solution for other tasks

**Comment by davydovbv** [comment]
I hope I have fixed everything)

File: build.gradle [hunk 10]
```diff
@@ -12,6 +12,9 @@ java {
 }
 
 dependencies {
+    implementation 'org.junit.jupiter:junit-jupiter:5.4.2'
+    implementation 'org.junit.jupiter:junit-jupiter:5.4.2'
+    implementation 'org.testng:testng:7.1.0'
```
Comment (rakovets) [inline build.gradle:17]: Don't touch configuration!!!

**Review by rakovets — APPROVED** [review]
Approved

**Pull request closed without merging**